tests/planner/ompl/test_OMPLSimplifier.cpp

```diff
@@ -0,0 +1,189 @@
+  aikido::constraint::TestablePtr boundsConstraint = aikido::constraint::createTestableBounds(stateSpace);
+  aikido::constraint::TestablePtr collConstraint = std::make_shared<MockTranslationalRobotConstraint>(
+        stateSpace, Eigen::Vector3d(-0.1, -0.1, -0.1),
+        Eigen::Vector3d(0.1, 0.1, 0.1));
+
+  auto simplifiedPair = aikido::planner::ompl::simplifyOMPL(
+      stateSpace, interpolator, std::move(dmetric), std::move(sampler), 
+      std::move(collConstraint), std::move(boundsConstraint), std::move(boundsProjection),
+      0.1, 5.0, 10, traj);
+
+  // Simplification results
+  auto simplifiedTraj = std::move(simplifiedPair.first);
+
+  // Check the first waypoint
+  auto s0 = stateSpace->createState();
+  simplifiedTraj->evaluate(0, s0);
+  auto r0 = s0.getSubStateHandle<R3>(0);
+  EXPECT_TRUE(r0.getValue().isApprox(startPose));
+
+  // Check the last waypoint
+  simplifiedTraj->evaluate(simplifiedTraj->getDuration(), s0);
+  r0 = s0.getSubStateHandle<R3>(0);
+  EXPECT_TRUE(r0.getValue().isApprox(goalPose));
+
+}
+
+
+// Test that the boolean returned is in agreement with shortening process
+TEST_F(SimplifierTest, ShortenThreeWayPointTraj)
+{
+
+  Eigen::Vector3d startPose(-5, -5, 0);
+  Eigen::Vector3d goalPose(5, 5, 0);
+
+  auto traj = planTrajForTest(stateSpace, interpolator, dmetric, sampler, collConstraint, 
+      boundsConstraint, boundsProjection, startPose, goalPose);
+
+  // Setup for simplification
+  aikido::distance::DistanceMetricPtr dmetric = aikido::distance::createDistanceMetric(stateSpace);
+  aikido::constraint::SampleablePtr sampler = aikido::constraint::createSampleableBounds(stateSpace, make_rng()); 
+  aikido::constraint::ProjectablePtr boundsProjection = aikido::constraint::createProjectableBounds(stateSpace);
+  aikido::constraint::TestablePtr boundsConstraint = aikido::constraint::createTestableBounds(stateSpace);
+  aikido::constraint::TestablePtr collConstraint = std::make_shared<MockTranslationalRobotConstraint>(
+        stateSpace, Eigen::Vector3d(-0.1, -0.1, -0.1),
+        Eigen::Vector3d(0.1, 0.1, 0.1));
+
+  auto simplifiedPair = aikido::planner::ompl::simplifyOMPL(
+      stateSpace, interpolator, std::move(dmetric), std::move(sampler), 
+      std::move(collConstraint), std::move(boundsConstraint), std::move(boundsProjection),
+      0.1, 5.0, 10, traj);
+
+  // Simplification results
+  auto simplifiedTraj = std::move(simplifiedPair.first);
+  bool shorten_success = simplifiedPair.second;
+
+  aikido::distance::DistanceMetricPtr Ddmetric = aikido::distance::createDistanceMetric(stateSpace);
+  double trajDistance = computeTrajLength<aikido::trajectory::InterpolatedPtr>(traj, stateSpace, Ddmetric);
+  double simplifiedTrajDistance = computeTrajLength<std::unique_ptr<aikido::trajectory::Interpolated>>(simplifiedTraj, stateSpace, Ddmetric);
+
+  EXPECT_TRUE(shorten_success == (simplifiedTrajDistance < trajDistance));  
```

Comment:
since I took 3 waypoints, I gave `maxEmptySteps = 20` with `timeOut = 20`. The probability of failing now should be less than 0.1%.